src/parquet/file_writer.h

```diff
@@ -102,6 +144,8 @@ class PARQUET_EXPORT ParquetFileWriter {
 
     virtual RowGroupWriter* AppendRowGroup() = 0;
 
+    virtual RowGroupWriter2* AppendRowGroup2() = 0;
+
```

Comment:
Consider a subclass for writing fixed size rowgroups that overrides `AppendRowGroup` or take a param in the `ParquetFileWriter` ctor that determines how it wants to pack rowgroups.  That way whatever client code is pushing tuples in doesn't have to care about the on-disk layout after the writer is created.

Reply:
Agreed. I think these details should be handled in the `WriterProperties` and the API of `RowGroupWriter` should be able to indicate to the caller whether it is "safe" to continue writing, or whether the current row group needs to be terminated and flushed

Reply:
This will restrict the clients from being able to loosely or tightly bound a RowGroup size.
Example: Clients might want to write a row at a time or a batch at a time or do both adaptively to achieve their target RowGroup size.